### PR TITLE
feat(ui): Change Select2Field to use react-select

### PR DESCRIPTION
**Pull request by billyvg**
* [x] requires a PR in getsentry to be deployed together
* [x] kill the old Account Notifications page before proceeding.

Depends on:
* ~~https://github.com/getsentry/sentry/pull/8688~~
* ~~https://github.com/getsentry/sentry/pull/8723~~
* https://github.com/getsentry/getsentry/pull/1961

### CODE REVIEW

**Comment by lynnagara** [comment]
This can be deployed before the getsentry PR right? Since you do export selectField as select2Field and the new API supports the old format.

File: src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx [hunk 1]
```diff
@@ -236,7 +236,8 @@ const RuleEditor = createReactClass({
             <AlertRuleRow>
               {t(
                 'Every time %s of these conditions are met:',
-                <Select2Field
+                <SelectField
+                  clearable={false}
```
Comment (lynnagara) [inline src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx]: Note that backspaceRemoves and deleteRemoves still default to true whereever you turn off clearable

File: package.json [hunk 2]
```diff
@@ -77,7 +77,6 @@
     "react-sparklines": "1.7.0",
     "reflux": "0.4.1",
     "scroll-to-element": "^2.0.0",
-    "select2": "3.5.1",
```
Comment (lynnagara) [inline package.json]: 👍

Comment (lynnagara) [reply package.json]: i think i recall a css file that we'd need to manually remove as well

File: src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx (hunk 1 above)
Comment (billyvg) [reply src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx]: Hmm should we make `backspaceRemoves/deleteRemoves` default to value of `clearable` in `<SelectControl>`? Then allow for it to be explicitly overridden.

File: package.json (hunk 2 above)
Comment (billyvg) [reply package.json]: Yeah it's removed (did I forget to push some updates?) https://github.com/getsentry/sentry/pull/8743/files#diff-ac24416e63ab1dd5bbc404fa65215557L7

**Comment by billyvg** [comment]
I think we'll need to sync this up with getsentry PR because we need to add the `react-select` CSS to getsentry. There's also some code changes (we were adding empty options to our `choices` props so it could have an "unselected" state).

File: package.json (hunk 2 above)
Comment (lynnagara) [reply package.json]: You're good, i missed it in the diff

File: src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx (hunk 1 above)
Comment (lynnagara) [reply src/sentry/static/sentry/app/views/settings/projectAlerts/ruleEditor/index.jsx]: Sounds good re: backspaceRemoves/deleteRemoves, lets leave the react select default for clearable

**Review by lynnagara — APPROVED** [review]
This PR looks good, though I'd slightly prefer to have the clearable default in line with the library to avoid one layer of misdirection. It would be ideal if we can add the new CSS to getsentry first to the PRs to avoid the dependency (but not sure how to avoid the empty option thing)

**Pull request merged**